PostHog/PostHogSDK.swift

```diff
@@ -478,28 +478,38 @@ let maxRetryDelay = 30.0
         capture(event, properties: nil, userProperties: nil, userPropertiesSetOnce: nil, groups: nil)
     }
 
-    @objc(captureWithEvent:properties:)
+    @objc(captureWithEvent:distinctId:)
```

Comment:
Can you check if this would play well with all the PostHogObjCExample?
The goal would be to not force people passing distinctId=nil in ObjC, ObjC does not handle well the optional, and they should not be forced to pass distinctId either.

Reply:
since this is a corner case, I'd be ok offering just one overload with all options at once and `distinctId`, for Swift this is not an issue, and for objC might be a bit verbose but this type of captured event is an edge case anyway.
So i'd rather focus the UX on the APIs that are used more often, for example:

```
    @objc(captureWithEvent:distinctId,properties:userProperties:userPropertiesSetOnce:groups:)
    public func capture(_ event: String,
                        distinctId: String? = nil,
                        properties: [String: Any]? = nil,
                        userProperties: [String: Any]? = nil,
                        userPropertiesSetOnce: [String: Any]? = nil,
                        groups: [String: String]? = nil)
```